agreement/demux.go

```diff
@@ -122,6 +123,13 @@ func (d *demux) tokenizeMessages(ctx context.Context, net Network, tag protocol.
 
 				o, err := tokenize(raw.Data)
 				if err != nil {
+					// check protocol version
+					cv, err := d.ledger.ConsensusVersion(d.ledger.NextRound())
+					if err != nil {
+						if _, ok := config.Consensus[cv]; !ok {
+							d.log.Warnf("received proposal with unsupported consensus version: %v", cv)
```

Comment:
The logic here looks correct. Two small tweaks:
1. if you're writing this new log message, then the other log message is no longer needed.
2. could you change the log message to something like 
d.log.Warnf("received proposal message was ignored. The node binary doesn't support the next network consensus (%v) and would no longer be able to process agreement messages", cv)